client/src/app/app-routing.module.ts

```diff
@@ -0,0 +1,20 @@
+import { NgModule } from '@angular/core';
+import { RouterModule, Routes } from '@angular/router';
+import { UserlistComponent } from './userlist/userlist.component';
+import { EditviewComponent } from './editview/editview.component';
+import { DashboardComponent } from './dashboard/dashboard.component';
+import { UserDetailComponent } from './user-detail/user-detail.component';
+
+const routes: Routes = [
+  { path: 'userlist', component: UserlistComponent },
+  { path: 'edit', component: EditviewComponent },
+  { path: 'dashboard', component: DashboardComponent },
+  { path: 'detail/:id', component: UserDetailComponent },
```

Comment:
Besides: You point with every route (except the dashboard) to a user-concerning component. Therefore, create at least a user-module and write something like this:

```suggestion
  { path: '', component: UserlistComponent },
  { path: 'edit/:id', component: EditviewComponent },
  { path: 'detail/:id', component: UserDetailComponent },
```